src/lib/init.js

```diff
@@ -36,49 +34,30 @@ function readExternalConsentData(config) {
 	});
 }
 
-function readInternalConsentData() {
-	return Promise.all([
-		[
-			readVendorConsentCookie(),
-			fetchPubVendorList(),
-			readPublisherConsentCookie()
-		]
-	]);
-}
-
-function readGlobalAndExternalConsentData(config) {
-	return Promise.all([
-		readExternalConsentData(config),
-		readVendorConsentCookie()
-	]);
-}
-
-export function init(configUpdates) {
+export function init (configUpdates) {
 	config.update(configUpdates);
 	log.debug('Using configuration:', config);
 
 	// Fetch the current vendor consent before initializing
-	return ((config.getConsentData) ? readGlobalAndExternalConsentData(config) : readInternalConsentData())
-		.then(([[vendorConsentData, pubVendorsList, publisherConsentData], globalVendorConsentData]) => {
-			const {vendors} = pubVendorsList || {};
+	return ((config.getConsentData) ? readExternalConsentData(config) : readConsentCookie())
+		.then((consentData) => {
+			const cmpApi = new CmpApi(CMP_ID, CMP_VERSION);
 
-			// Check config for allowedVendorIds then the pubVendorList
-			const {allowedVendorIds: configVendorIds} = config;
-			const allowedVendorIds = configVendorIds instanceof Array && configVendorIds.length ? configVendorIds :
-				vendors && vendors.map(vendor => vendor.id);
+			if (config.decoratePageCallHandler) {
+				config.decoratePageCallHandler(CmpApi, cmpApi);
```

Comment:
Do we need to pass `CmpApi` class here? It could be useful when we would like to decorate all future instances of this class. But new instance of `CmpApi` is created only once - few lines above.

Reply:
As far as I understood Artur's needs this code is necessary.

Reply:
I've just talk to Arthur and he doesn't need it :)